zerver/actions/realm_settings.py

```diff
@@ -233,13 +233,29 @@ def get_realm_authentication_methods_for_page_params_api(
     # The rest of the function is only for the mechanism of restricting
     # certain backends based on the realm's plan type on Zulip Cloud.
 
+    from corporate.models import CustomerPlan
+
     for backend_name in result_dict:
         available_for = AUTH_BACKEND_NAME_MAP[backend_name].available_for_cloud_plans
         if available_for is not None and realm.plan_type not in available_for:
             result_dict[backend_name]["available"] = False
-            result_dict[backend_name]["unavailable_reason"] = _(
-                "Not available on your current plan"
+
+            required_upgrade_plan_number = min(
+                set(available_for).intersection({Realm.PLAN_TYPE_STANDARD, Realm.PLAN_TYPE_PLUS})
             )
+            if required_upgrade_plan_number == Realm.PLAN_TYPE_STANDARD:
+                required_upgrade_plan_name = CustomerPlan.name_from_tier(
+                    CustomerPlan.TIER_CLOUD_STANDARD
+                )
```

Comment:
It feels like a significant code smell that our `realm.plan_type` value don't map to `CustomerPlan.tier` values in any nice way

Reply:
Yeah they agree for the remote plan types but not for `Realm`. We can probably do a migration to line the up better, but in any case, this sort of code is the system we want for readability reasons anyway, I think.